mmaction/models/backbones/resnet_audio.py

```diff
@@ -0,0 +1,374 @@
+            layer is the 3x3 conv layer, otherwise the stride-two layer is
+            the first 1x1 conv layer. Default: 'pytorch'.
+        factorize (bool): Whether to factorize kernel. Default: True.
+        factorize_style (str): `3x1x1` or `1x1x1`. which determines the kernel
+            sizes and padding strides for conv1 and conv2 in each block.
+            Default: '3x1x1'.
+        norm_cfg (dict): Config for norm layers. required keys are `type`,
+            Default: dict(type='BN3d').
+        with_cp (bool): Use checkpoint or not. Using checkpoint will save some
+            memory while slowing down the training speed. Default: False.
+    """
+    expansion = 4
+
+    def __init__(self,
+                 inplanes,
+                 planes,
+                 stride=2,
+                 dilation=1,
+                 downsample=None,
+                 style='pytorch',
+                 factorize=True,
+                 norm_cfg=dict(type='SyncBN'),
+                 with_cp=False):
+        super().__init__()
+
+        self.inplanes = inplanes
+        self.planes = planes
+        self.stride = stride
+        self.dilation = dilation
+        self.style = style
+        self.factorize = factorize
+        self.norm_cfg = norm_cfg
+        self.with_cp = with_cp
+
+        self.conv1_stride = 1
+        self.conv2_stride = stride
+
+        conv1_kernel_size = (1, 1)
+        conv1_padding = 0
+        conv2_kernel_size = (3, 3)
+        conv2_padding = (dilation, dilation)
+        self.conv1 = ConvModule(
+            inplanes,
+            planes,
+            kernel_size=conv1_kernel_size,
+            padding=conv1_padding,
+            dilation=dilation,
+            bias=False)
+        self.conv2 = ConvModule(
+            planes,
+            planes,
+            kernel_size=conv2_kernel_size,
+            stride=stride,
+            padding=conv2_padding,
+            dilation=dilation,
+            bias=False,
+            conv_cfg=dict(type='ConvAudio') if factorize else dict(
+                type='Conv'))
+        self.conv3 = ConvModule(
+            planes,
+            planes * self.expansion,
+            kernel_size=1,
+            bias=False,
+            act_cfg=None)
+
+        self.relu = nn.ReLU(inplace=True)
+        self.downsample = downsample
+
+    def forward(self, x):
+
+        def _inner_forward(x):
+            identity = x
+            out = self.conv1(x)
+            out = self.conv2(out)
+            out = self.conv3(out)
+
+            if self.downsample is not None:
+                identity = self.downsample(x)
+            out += identity
+
+            return out
+
+        if self.with_cp and x.requires_grad:
+            out = cp.checkpoint(_inner_forward, x)
+        else:
+            out = _inner_forward(x)
+
+        out = self.relu(out)
+
+        return out
+
+
+@BACKBONES.register_module
+class ResNet2dAudio(nn.Module):
```

Comment:
How about inheriting from `ResNet`

Reply:
Yeah, I have considered it. Actually, it is more like a ResNet3D. It needs to factorize all conv2ds in the first two res-stages, which is similar to ResNet3D(which inflates some stages). Almost all functions should be overrides including BasicBlock/Bottleneck, make_res_stage, init_weight. I think it's better to write it independently like ResNet3D for clarity.